pkg/apis/pipeline/v1alpha1/condition_validation.go

```diff
@@ -41,5 +41,13 @@ func (cs *ConditionSpec) Validate(ctx context.Context) *apis.FieldError {
 	if cs.Check.Image == "" {
 		return apis.ErrMissingField("Check.Image")
 	}
+	if cs.Check.Script != "" {
```

Comment:
If you are feeling motivated, it might be worth seeing if we can reuse the `validateSteps` function from `task_validation.go` here as well: https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1alpha1/task_validation.go#L169

Reply:
I reused it "as is",  validation error is not ideal, as it mentions steps - up to you if we want it improve it somehow